### PR TITLE
Update eslint configuration

### PR DESCRIPTION
**Pull request by Swastikyadav**
- Current Eslint configuration forces to use regular javascript functions.
- Arrow functions don't work.
- Changed Eslint configuration to allow arrow functions.
- Added babel-eslint in package.json.
- Removed eslint rules that conflict with prettier.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -62,7 +62,7 @@ gem "puma", "~> 3.12"
 
 # Rails request timeout, needed if running on Heroku-
 # https://devcenter.heroku.com/articles/request-timeout
-gem "rack-timeout"
+gem "rack-timeout", require: "rack/timeout/base"
```
Comment (neerajdotname) [inline Gemfile:65]: @Swastikyadav This change has nothing to do with `eslint`. Please remove it from this PR and put in a different PR.

Comment (Swastikyadav) [reply Gemfile]: I have removed the unnecessary change in the gemfile. We can merge this PR now.

**Comment by neerajdotname** [comment]
@Swastikyadav _a Could you please rebase this PR with master. And then assign the PR to me.

**Comment by Swastikyadav** [comment]
@neerajdotname _a I have rebased this branch with master.

**Comment by neerajdotname** [comment]
@Swastikyadav You mentioned `Arrow functions don't work.`. Could you please expand on that. What happens if one uses arrow functions.

**Comment by Swastikyadav** [comment]
> @Swastikyadav You mentioned `Arrow functions don't work.`. Could you please expand on that. What happens if one uses arrow functions.

After writing arrow function, If you try to commit your changes `eslint` will give this error:

`error  Parsing error: Unexpected token =`

**Comment by neerajdotname** [comment]
@vinay0x _a Can you review this.

File: package.json [hunk 2]
```diff
@@ -30,6 +30,7 @@
   "dependencies": {
     "@babel/preset-react": "^7.0.0",
     "@rails/webpacker": "^4.0.2",
+    "babel-eslint": "^10.1.0",
```
Comment (karansapolia) [inline package.json:33]: As it is used only at build time, I think it will be better if we keep babel-eslint in `devDependencies`

**Comment by Swastikyadav** [comment]
@karansapolia _a Thanks for such a brief comment. I made the requested changes, please have a look.

**Comment by karansapolia** [comment]
@Swastikyadav LGTM 👍. Thanks for making the requested changes.

**Comment by karansapolia** [comment]
Resolves: #331 

@Swastikyadav It would be great to always include `fix: #<issue number>`or `resolve: #<issue number>` in commit messages. That helps keep a track of what new changes/commits have been made for an issue on the issue's page, and also helps automatically close an issue once a PR for it has been merged. I forgot mentioning the issue number here. 

Some resources I found helpful in writing good commit messages are:
https://webhint.io/docs/contributor-guide/getting-started/pull-requests/#commit-messages
https://www.conventionalcommits.org/en/v1.0.0/

We don't follow the `<Tag>: Capitalized, summary (50 chars recommended)` pattern but we do mention the issue our commit is made for in the last line.
